lib/smf/smf.c

```diff
@@ -9,24 +9,24 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(smf);
 
-/*
- * Private structure (to this file) used to track state machine context.
- * The structure is not used directly, but instead to cast the "internal"
- * member of the smf_ctx structure.
+/**
+ * @brief Private structure (to this file) used to track state machine context.
+ *        The structure is not used directly, but instead to cast the "internal"
+ *        member of the smf_ctx structure.
  */
 struct internal_ctx {
-	bool new_state : 1;
-	bool terminate : 1;
-	bool exit      : 1;
-	bool handled   : 1;
+	bool new_state: 1;
+	bool terminate: 1;
+	bool exit: 1;
+	bool handled: 1;
+	bool entry: 1;
 };
 
-static bool share_paren(const struct smf_state *test_state,
-			const struct smf_state *target_state)
+#if CONFIG_SMF_ANCESTOR_SUPPORT
```

Comment:
Because you're using the `__unused` attribute, wrapping these functions in an `#ifdef` isn't required.

Reply:
The `#ifdef` is required because we conditionally compile in `parent` in `smf_state`

Should I delete the `__unused`?

Reply:
Yes.

Reply:
OK, Done.

Reply:
> ~Yes.~ Sorry - I responded before looking at the latest patchset. You need either `__unused` or `#ifdef`. I'm good with the `__unused`.

Unfortunately you need the `#ifdefs`, so I'll delete the `__unused`